### PR TITLE
cdc: make mq encoding protocol related more clear

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v8.4 (TiDB 8.4 versions)
- [x] v8.3 (TiDB 8.3 versions)
- [x] v8.2 (TiDB 8.2 versions)
- [x] v8.1 (TiDB 8.1 versions)
- [x] v7.5 (TiDB 7.5 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [x] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: ticdc/ticdc-avro-protocol.md [hunk 1]
```diff
@@ -266,14 +272,27 @@ DECIMAL(10, 4)
 
 ## DDL 事件与 Schema 变更
 
-Avro 并不会向下游生成 DDL 事件。Avro 会在每次 DML 事件发生时检测是否发生 schema 变更，如果发生了 schema 变更，Avro 会生成新的 schema，并尝试向 Schema Registry 注册。注册时，Schema Registry 会做兼容性检测，如果此次 schema 变更没有通过兼容性检测，注册将会失败，TiCDC 并不会尝试解决 schema 的兼容性问题。
+Avro 协议并不会向下游发送 DDL 事件和 Watermark 事件。Avro 会在每次 DML 事件发生时检测是否发生 schema 变更，如果发生了 schema 变更，Avro 会生成新的 schema，并尝试向 Schema Registry 注册。注册时，Schema Registry 会做兼容性检测，如果此次 schema 变更没有通过兼容性检测，注册将会失败，TiCDC 并不会尝试解决 schema 的兼容性问题。
 
-同时，即使 schema 变更通过兼容性检测并成功注册新版本，数据的生产者和消费者可能仍然需要升级才能正确工作。
+同时，即使 schema 变更通过兼容性检测并成功注册新版本，数据的生产者和消费者可能仍然需要获取到新版本的 schema，才能对数据进行正确解码。
```
Comment (lidezhu) [inline ticdc/ticdc-avro-protocol.md:277]: 这一句的生产者应该去掉？生产者需要解码吗？

**Review by lidezhu — APPROVED** [review]
LGTM

File: ticdc/ticdc-avro-protocol.md [hunk 2]
```diff
@@ -5,52 +5,34 @@ summary: 了解 TiCDC Avro Protocol 的概念和使用方法。
...
 使用 Avro 时的配置样例如下所示：
 
 ```shell
 cdc cli changefeed create --server=http://127.0.0.1:8300 --changefeed-id="kafka-avro" --sink-uri="kafka://127.0.0.1:9092/topic-name?protocol=avro" --schema-registry=http://127.0.0.1:8081 --config changefeed_config.toml
 ```
 
+`--schema-registry` 的值支持 https 协议和 username:password 认证，比如`--schema-registry=https://username:password@schema-registry-uri.com`，username 和 password 必须经过 URL 编码。
+
+> **注意：**
+> 使用 Avro 协议时，一个 Kafka Topic 必须只能有一张表的数据，需要在 config 文件中配置 [Topic 分发器](/ticdc/ticdc-sink-to-kafka.md#topic-分发器)。
```
Comment (hfxsd) [inline ticdc/ticdc-avro-protocol.md:27]:
```suggestion
> **注意：**
> 
> 使用 Avro 协议时，一个 Kafka Topic 必须只能有一张表的数据，需要在 config 文件中配置 [Topic 分发器](/ticdc/ticdc-sink-to-kafka.md#topic-分发器)。 
```

File: ticdc/ticdc-sink-to-kafka.md [hunk 3]
```diff
@@ -180,7 +181,7 @@ dispatchers = [
 
 ### TiCDC 集成 AWS Glue Schema Registry
 
-从 v7.4.0 开始，TiCDC 支持在用户选择 Avro 协议同步数据时使用 [AWS Glue Schema Registry](https://docs.aws.amazon.com/glue/latest/dg/schema-registry.html) 作为 Schema Registry。配置样例如下所示：
+从 v7.4.0 开始，TiCDC 支持在用户选择  [Avro 协议](/ticdc/ticdc-avro-protocol.md) 同步数据时使用 [AWS Glue Schema Registry](https://docs.aws.amazon.com/glue/latest/dg/schema-registry.html) 作为 Schema Registry。配置样例如下所示：
```
Comment (hfxsd) [inline ticdc/ticdc-sink-to-kafka.md:184]:
```suggestion
从 v7.4.0 开始，TiCDC 支持在用户选择  [Avro 协议](/ticdc/ticdc-avro-protocol.md)同步数据时使用 [AWS Glue Schema Registry](https://docs.aws.amazon.com/glue/latest/dg/schema-registry.html) 作为 Schema Registry。配置样例如下所示：
```

File: ticdc/ticdc-sink-to-kafka.md [hunk 4]
```diff
@@ -161,7 +162,7 @@ URI 中可配置的的参数如下：
 
 ### TiCDC 集成 Kafka Connect (Confluent Platform)
 
-如要使用 Confluent 提供的 [data connectors](https://docs.confluent.io/current/connect/managing/connectors.html) 向关系型或非关系型数据库传输数据，请选择 `avro` 协议，并在 `schema-registry` 中提供 [Confluent Schema Registry](https://www.confluent.io/product/confluent-platform/data-compatibility/) 的 URL。
+如要使用 Confluent 提供的 [data connectors](https://docs.confluent.io/current/connect/managing/connectors.html) 向关系型或非关系型数据库传输数据，请选择 [Avro 协议](/ticdc/ticdc-avro-protocol.md) 协议，并在 `schema-registry` 中提供 [Confluent Schema Registry](https://www.confluent.io/product/confluent-platform/data-compatibility/) 的 URL。
```
Comment (hfxsd) [inline ticdc/ticdc-sink-to-kafka.md:165]:
```suggestion
如要使用 Confluent 提供的 [data connectors](https://docs.confluent.io/current/connect/managing/connectors.html) 向关系型或非关系型数据库传输数据，请选择 [Avro 协议](/ticdc/ticdc-avro-protocol.md)协议，并在 `schema-registry` 中提供 [Confluent Schema Registry](https://www.confluent.io/product/confluent-platform/data-compatibility/) 的 URL。
```

File: ticdc/ticdc-sink-to-kafka.md [hunk 5]
```diff
@@ -73,14 +73,14 @@ URI 中可配置的的参数如下：
 | `127.0.0.1`          | 下游 Kafka 对外提供服务的 IP。                                 |
 | `9092`               | 下游 Kafka 的连接端口。                                          |
 | `topic-name`         | 变量，使用的 Kafka topic 名字。                                      |
+| `protocol`           | 输出到 Kafka 的消息协议。可选值有 [canal-json](/ticdc/ticdc-canal-json.md)、[open-protocol](/ticdc/ticdc-open-protocol.md)、[avro](/ticdc/ticdc-avro-protocol.md)、[debezium](/ticdc/ticdc-debezium.md) 和 [simple](/ticdc/ticdc-simple-protocol.md)。 |
```
Comment (hfxsd) [inline ticdc/ticdc-sink-to-kafka.md:76]:
```suggestion
| `protocol`           | 输出到 Kafka 的消息协议。可选值有 [`canal-json`](/ticdc/ticdc-canal-json.md)、[`open-protocol`](/ticdc/ticdc-open-protocol.md)、[`avro`](/ticdc/ticdc-avro-protocol.md)、[`debezium`](/ticdc/ticdc-debezium.md) 和 [`simple`](/ticdc/ticdc-simple-protocol.md)。 |
```

File: ticdc/ticdc-avro-protocol.md [hunk 6]
```diff
@@ -266,14 +272,27 @@ DECIMAL(10, 4)
...
+比如，Confluent Schema Registry 默认的[兼容性策略](https://docs.confluent.io/platform/current/schema-registry/fundamentals/schema-evolution.html#compatibility-types)是 BACKWARD，在这种策略下，如果你在源表增加一个非空列，Avro 在生成新 schema 向 Schema Registry 注册时将会因为兼容性问题失败，这个时候 changefeed 将会进入 error 状态。
 
-比如，Confluent Schema Registry 默认的兼容性策略是 BACKWARD，在这种策略下，如果你在源表增加一个非空列，Avro 在生成新 schema 向 Schema Registry 注册时将会因为兼容性问题失败，这个时候 changefeed 将会进入 error 状态。
+同时，即使 schema 变更通过兼容性检测并成功注册新版本，数据的生产者和消费者可能仍然需要获取到新版本的 schema，才能对数据进行正确编解码。
 
 如需了解更多 schema 相关信息，请参阅 [Schema Registry 的相关文档](https://docs.confluent.io/platform/current/schema-registry/avro.html)。
 
+## 消费者实现
+
+TiCDC Avro 协议，支持被 [io.confluent.kafka.serializers.KafkaAvroDeserializer](https://docs.confluent.io/platform/current/schema-registry/fundamentals/serdes-develop/serdes-avro.html#avro-deserializer) 反序列化。
```
Comment (hfxsd) [inline ticdc/ticdc-avro-protocol.md:285]:
```suggestion
TiCDC Avro 协议，支持被 [`io.confluent.kafka.serializers.KafkaAvroDeserializer`](https://docs.confluent.io/platform/current/schema-registry/fundamentals/serdes-develop/serdes-avro.html#avro-deserializer) 反序列化。
```

File: ticdc/ticdc-avro-protocol.md [hunk 7]
```diff
@@ -266,14 +272,27 @@ DECIMAL(10, 4)
...
+
+消费者程序可以通过 [Schema Registry API](https://docs.confluent.io/platform/current/schema-registry/develop/api.html) 获取到最新的 schema，然后对数据进行反序列化。
+
+### 区分事件类型
+
+消费者程序可以按照如下规则区分 DML 事件类型：
+
+* 只有 Key 部分，则是 Delete 事件。
+* 含有 Value 部分，则是 Insert 或 Update 事件。如果用户开启了 TiDB 扩展字段功能，可以根据其中的 `_tidb_op` 字段，判断该条事件变更是 Insert 或 Update。反之无法区分。
```
Comment (hfxsd) [inline ticdc/ticdc-avro-protocol.md:294]:
```suggestion
* 含有 Value 部分，则是 Insert 或 Update 事件。如果用户开启了 [TiDB 扩展字段功能](#tidb-扩展字段)，可以根据其中的 `_tidb_op` 字段，判断该条事件变更是 Insert 或 Update。反之无法区分。
```

**Review by hfxsd — APPROVED** [review]

File: ticdc/ticdc-avro-protocol.md [hunk 8]
```diff
@@ -5,52 +5,35 @@ summary: 了解 TiCDC Avro Protocol 的概念和使用方法。
...
 
 ```shell
 cdc cli changefeed create --server=http://127.0.0.1:8300 --changefeed-id="kafka-avro" --sink-uri="kafka://127.0.0.1:9092/topic-name?protocol=avro" --schema-registry=http://127.0.0.1:8081 --config changefeed_config.toml
 ```
 
+`--schema-registry` 的值支持 https 协议和 username:password 认证，比如`--schema-registry=https://username:password@schema-registry-uri.com`，username 和 password 必须经过 URL 编码。
+
+> **注意：**
+> 
+> 使用 Avro 协议时，一个 Kafka Topic 必须只能有一张表的数据，需要在 config 文件中配置 [Topic 分发器](/ticdc/ticdc-sink-to-kafka.md#topic-分发器)。
```
Comment (hfxsd) [inline ticdc/ticdc-avro-protocol.md:28]:
```suggestion
> 使用 Avro 协议时，一个 Kafka Topic 必须只能有一张表的数据。你需要在配置文件中配置 [Topic 分发器](/ticdc/ticdc-sink-to-kafka.md#topic-分发器)。 
```

File: ticdc/ticdc-avro-protocol.md [hunk 9]
```diff
@@ -155,7 +162,7 @@ Column 数据格式即 Key/Value 数据格式中的 `{{ColumnValueBlock}}` 部
 
 - `{{ColumnName}}` 表示列名。
 - `{{TIDB_TYPE}}` 表示对应到 TiDB 中的类型，与原始的 SQL Type 不是一一对应关系。
-- `{{AVRO_TYPE}}` 表示 [avro spec](https://avro.apache.org/docs/current/spec.html) 中的类型。
+- `{{AVRO_TYPE}}` 表示 [avro spec](https://avro.apache.org/docs/++version++/specification) 中的类型。
```
Comment (hfxsd) [inline ticdc/ticdc-avro-protocol.md:165]:
```suggestion
- `{{AVRO_TYPE}}` 表示 [Avro Specification](https://avro.apache.org/docs/++version++/specification) 中的类型。
```

File: ticdc/ticdc-avro-protocol.md [hunk 10]
```diff
@@ -266,14 +273,27 @@ DECIMAL(10, 4)
 
 ## DDL 事件与 Schema 变更
 
-Avro 并不会向下游生成 DDL 事件。Avro 会在每次 DML 事件发生时检测是否发生 schema 变更，如果发生了 schema 变更，Avro 会生成新的 schema，并尝试向 Schema Registry 注册。注册时，Schema Registry 会做兼容性检测，如果此次 schema 变更没有通过兼容性检测，注册将会失败，TiCDC 并不会尝试解决 schema 的兼容性问题。
+Avro 协议并不会向下游发送 DDL 事件和 Watermark 事件。Avro 会在每次 DML 事件发生时检测是否发生 schema 变更，如果发生了 schema 变更，Avro 会生成新的 schema，并尝试向 Schema Registry 注册。注册时，Schema Registry 会做兼容性检测，如果此次 schema 变更没有通过兼容性检测，注册将会失败，TiCDC 并不会尝试解决 schema 的兼容性问题。
 
-同时，即使 schema 变更通过兼容性检测并成功注册新版本，数据的生产者和消费者可能仍然需要升级才能正确工作。
+比如，Confluent Schema Registry 默认的[兼容性策略](https://docs.confluent.io/platform/current/schema-registry/fundamentals/schema-evolution.html#compatibility-types)是 BACKWARD，在这种策略下，如果你在源表增加一个非空列，Avro 在生成新 schema 向 Schema Registry 注册时将会因为兼容性问题失败，这个时候 changefeed 将会进入 error 状态。
```
Comment (hfxsd) [inline ticdc/ticdc-avro-protocol.md:278]:
```suggestion
比如，Confluent Schema Registry 默认的[兼容性策略](https://docs.confluent.io/platform/current/schema-registry/fundamentals/schema-evolution.html#compatibility-types)是 `BACKWARD`，在这种策略下，如果你在源表增加一个非空列，Avro 在生成新 schema 向 Schema Registry 注册时将会因为兼容性问题失败，这个时候 changefeed 将会进入 error 状态。
```

File: ticdc/ticdc-avro-protocol.md [hunk 11]
```diff
@@ -266,14 +273,27 @@ DECIMAL(10, 4)
...
 
-比如，Confluent Schema Registry 默认的兼容性策略是 BACKWARD，在这种策略下，如果你在源表增加一个非空列，Avro 在生成新 schema 向 Schema Registry 注册时将会因为兼容性问题失败，这个时候 changefeed 将会进入 error 状态。
+同时，即使 schema 变更通过兼容性检测并成功注册新版本，数据的生产者和消费者可能仍然需要获取到新版本的 schema，才能对数据进行正确编解码。
 
 如需了解更多 schema 相关信息，请参阅 [Schema Registry 的相关文档](https://docs.confluent.io/platform/current/schema-registry/avro.html)。
 
+## 消费者实现
+
+TiCDC Avro 协议，支持被 [`io.confluent.kafka.serializers.KafkaAvroDeserializer`](https://docs.confluent.io/platform/current/schema-registry/fundamentals/serdes-develop/serdes-avro.html#avro-deserializer) 反序列化。
```
Comment (hfxsd) [inline ticdc/ticdc-avro-protocol.md:286]:
```suggestion
TiCDC Avro 协议支持被 [`io.confluent.kafka.serializers.KafkaAvroDeserializer`](https://docs.confluent.io/platform/current/schema-registry/fundamentals/serdes-develop/serdes-avro.html#avro-deserializer) 反序列化。
```

File: ticdc/ticdc-avro-protocol.md [hunk 12]
```diff
@@ -5,52 +5,35 @@ summary: 了解 TiCDC Avro Protocol 的概念和使用方法。
...
 
 ```shell
 cdc cli changefeed create --server=http://127.0.0.1:8300 --changefeed-id="kafka-avro" --sink-uri="kafka://127.0.0.1:9092/topic-name?protocol=avro" --schema-registry=http://127.0.0.1:8081 --config changefeed_config.toml
 ```
 
+`--schema-registry` 的值支持 https 协议和 username:password 认证，比如`--schema-registry=https://username:password@schema-registry-uri.com`，username 和 password 必须经过 URL 编码。
+
+> **注意：**
+> 
+> 使用 Avro 协议时，一个 Kafka Topic 必须只能有一张表的数据。你需要在配置文件中配置 [Topic 分发器](/ticdc/ticdc-sink-to-kafka.md#topic-分发器)。
```
Comment (hfxsd) [inline ticdc/ticdc-avro-protocol.md:28]:
```suggestion
> 使用 Avro 协议时，一个 Kafka Topic 只能有一张表的数据。你需要在配置文件中配置 [Topic 分发器](/ticdc/ticdc-sink-to-kafka.md#topic-分发器)。 
```

File: ticdc/ticdc-sink-to-kafka.md [hunk 13]
```diff
@@ -109,12 +109,13 @@ URI 中可配置的的参数如下：
...
 * TiCDC 推荐用户自行创建 Kafka Topic，你至少需要设置该 Topic 每次向 Kafka broker 发送消息的最大数据量和下游 Kafka partition 的数量。在创建 changefeed 的时候，这两项设置分别对应 `max-message-bytes` 和 `partition-num` 参数。
-* 如果你在创建 changefeed 时，使用了尚未存在的 Topic，那么 TiCDC 会尝试使用 `partition-num` 和 `replication-factor` 参数自行创建 Topic。建议明确指定这两个参数。
+* 如果你在创建 changefeed 时，使用了尚未存在的 Topic，那么 TiCDC 会尝试使用 `partition-num` 和 `replication-factor` 参数自行创建 Topic，建议明确指定这两个参数。
 * 在大多数情况下，建议使用 `canal-json` 协议。
 
 > **注意：**
 >
-> 当 `protocol` 为 `open-protocol` 时，TiCDC 会尽量避免产生长度超过 `max-message-bytes` 的消息。但如果单条数据变更记录需要超过 `max-message-bytes` 个字节来表示，为了避免静默失败，TiCDC 会试图输出这条消息并在日志中输出 Warning。
+> 当 `protocol` 为 `open-protocol` 时，TiCDC 会将多个事件编码到同一个 Kafka 消息中，此过程中尽量避免产生长度超过 `max-message-bytes` 的消息。
+> 如果单条数据变更编码得到的消息大小超过了 `max-message-bytes` 个字节，changefeed 会报错，打印错误日志。
```
Comment (hfxsd) [inline ticdc/ticdc-sink-to-kafka.md:118]:
```suggestion
> 如果单条数据变更编码得到的消息大小超过了 `max-message-bytes` 个字节，changefeed 会报错，并打印错误日志。
```

File: ticdc/ticdc-avro-protocol.md [hunk 14]
```diff
@@ -89,7 +72,35 @@ Key 中的 `fields` 只包含主键或唯一索引列。
 }
 ```
 
-Value 数据格式默认与 Key 数据格式相同，但是 Value 的 `fields` 中包含了所有的列，而不仅仅是主键列。
+Value 数据格式默认与 Key 数据格式相同，但是 Value 的 `fields` 中包含了所有的列。
+
+> **注意：**
+> 
+> Avro 协议在编码 DML 事件时，操作方式如下：
+> - 对于 Delete 事件，只编码 Key 部分，Value 部分为空。
```
Comment (hfxsd) [inline ticdc/ticdc-avro-protocol.md:80]:
```suggestion
> Avro 协议在编码 DML 事件时，操作方式如下：
> 
> - 对于 Delete 事件，只编码 Key 部分，Value 部分为空。
```

File: ticdc/ticdc-avro-protocol.md [hunk 15]
```diff
@@ -5,52 +5,35 @@ summary: 了解 TiCDC Avro Protocol 的概念和使用方法。
...
 
-当使用 Message Queue (MQ) 作为下游 Sink 时，你可以在 `sink-uri` 中指定使用 Avro。TiCDC 获取 TiDB 的 DML 事件，并将这些事件封装到 Avro Message，然后发送到下游。当 Avro 检测到 schema 变化时，会向 Schema Registry 注册最新的 schema。
-
 使用 Avro 时的配置样例如下所示：
 
 ```shell
 cdc cli changefeed create --server=http://127.0.0.1:8300 --changefeed-id="kafka-avro" --sink-uri="kafka://127.0.0.1:9092/topic-name?protocol=avro" --schema-registry=http://127.0.0.1:8081 --config changefeed_config.toml
 ```
 
+`--schema-registry` 的值支持 https 协议和 username:password 认证，比如`--schema-registry=https://username:password@schema-registry-uri.com`，username 和 password 必须经过 URL 编码。
```
Comment (qiancai) [inline ticdc/ticdc-avro-protocol.md:24]:
```suggestion
`--schema-registry` 的值支持 HTTPS 协议和 username:password 认证，username 和 password 必须经过 URL 编码。例如，`--schema-registry=https://username:password@schema-registry-uri.com`。
```

File: ticdc/ticdc-avro-protocol.md [hunk 16]
```diff
@@ -89,7 +72,36 @@ Key 中的 `fields` 只包含主键或唯一索引列。
...
-Value 数据格式默认与 Key 数据格式相同，但是 Value 的 `fields` 中包含了所有的列，而不仅仅是主键列。
+Value 数据格式默认与 Key 数据格式相同，但是 Value 的 `fields` 中包含了所有的列。
+
+> **注意：**
+> 
+> Avro 协议在编码 DML 事件时，操作方式如下：
+> 
+> - 对于 Delete 事件，只编码 Key 部分，Value 部分为空。
+> - 对于 Insert 事件，编码所有列数据到 Value 部分。
+> - 对于 Update 事件，只编码更新后的所有列数据。
```
Comment (qiancai) [inline ticdc/ticdc-avro-protocol.md:83]:
```suggestion
> - 对于 Update 事件，只编码更新后的所有列数据到 Value 部分。
```

File: ticdc/ticdc-sink-to-kafka.md [hunk 17]
```diff
@@ -109,12 +109,13 @@ URI 中可配置的的参数如下：
...
 
 * TiCDC 推荐用户自行创建 Kafka Topic，你至少需要设置该 Topic 每次向 Kafka broker 发送消息的最大数据量和下游 Kafka partition 的数量。在创建 changefeed 的时候，这两项设置分别对应 `max-message-bytes` 和 `partition-num` 参数。
-* 如果你在创建 changefeed 时，使用了尚未存在的 Topic，那么 TiCDC 会尝试使用 `partition-num` 和 `replication-factor` 参数自行创建 Topic。建议明确指定这两个参数。
+* 如果你在创建 changefeed 时，使用了尚未存在的 Topic，那么 TiCDC 会尝试使用 `partition-num` 和 `replication-factor` 参数自行创建 Topic，建议明确指定这两个参数。
 * 在大多数情况下，建议使用 `canal-json` 协议。
 
 > **注意：**
 >
-> 当 `protocol` 为 `open-protocol` 时，TiCDC 会尽量避免产生长度超过 `max-message-bytes` 的消息。但如果单条数据变更记录需要超过 `max-message-bytes` 个字节来表示，为了避免静默失败，TiCDC 会试图输出这条消息并在日志中输出 Warning。
+> 当 `protocol` 为 `open-protocol` 时，TiCDC 会将多个事件编码到同一个 Kafka 消息中，此过程中尽量避免产生长度超过 `max-message-bytes` 的消息。
```
Comment (qiancai) [inline ticdc/ticdc-sink-to-kafka.md:117]:
```suggestion
> 当 `protocol` 为 `open-protocol` 时，TiCDC 会将多个事件编码到同一个 Kafka 消息中，并尽量避免在此过程中生成长度超过 `max-message-bytes` 的消息。
```

**Review by qiancai — APPROVED** [review]
Rest LGTM

File: ticdc/ticdc-avro-protocol.md [hunk 18]
```diff
@@ -5,52 +5,35 @@ summary: 了解 TiCDC Avro Protocol 的概念和使用方法。
...
+> **注意：**
+> 
+> 使用 Avro 协议时，一个 Kafka Topic 只能包含一张表的数据。你需要在配置文件中配置 [Topic 分发器](/ticdc/ticdc-sink-to-kafka.md#topic-分发器)。 
+
 ```shell
 [sink]
 dispatchers = [
  {matcher = ['*.*'], topic = "tidb_{schema}_{table}"},
 ]
 ```
```
Comment (qiancai) [inline ticdc/ticdc-avro-protocol.md:35]: L30~L35 在英文 PR https://github.com/pingcap/docs/pull/19156/files 里删掉了，中文里也要删除吗
image

Comment (hfxsd) [reply ticdc/ticdc-avro-protocol.md:35]: 没删吧，https://github.com/pingcap/docs/pull/19156/files#diff-b0deb5cbc83f533611f6f7495df8e86191a1ad8228719a288350b8c83c6ce519R32

Comment (3AceShowHand) [reply ticdc/ticdc-avro-protocol.md:35]: 这个不用删除的，给个例子总是好的。

**Pull request merged**